### DISCUSSION TITLE
Products by Category block: no pagination followup

### DISCUSSION THREAD

**Discussion by kentsor** [woocommerce/woocommerce#45045] [WooCommerce, Blocks, and Site Editing]
## Is your feature request related to a problem? Please describe.
This is following up to issue 2656, which was closed by saying that the blocks were deliberately designed not to have pagination.

I'm really not understanding that. Pagination and being able to display an unlimited number of results is indispensable. The block is _useless_ (at least to me) if it cannot display more than a screenful of products. In fact it's worse than useless, it's wrong because there is no indication of truncation of the results.

In what context does this crippling make any sense at all ?

**Comment by senadir** [comment]
Hey there @kentsor 

> In what context does this crippling make any sense at all ?

As Darren said in the original issue, we're not trying to replace the archive loop with a block, this block is intended to be a drop-in block in a larger page, and so pagination for it doesn't make sense.

Not only that, but we already have plans for building other blocks that are going to replace the archive loop and contain pagination, so your use case is going to get handled somewhere in the future, but it won't be above that block, but in a new block that is based on All Products.

It's not like we can toggle pagination and choose not to, implementing pagination is going to take a lot of dev time that we can put into a better solution or other higher priority features.
